samples/tutorials/mnist/04_Reusable_and_Pre-build_Components_as_Pipeline.ipynb

```diff
@@ -0,0 +1,799 @@
+   "metadata": {},
+   "source": [
+    "## Writing the program code"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The following cell creates a file `app.py` that contains a Python script. The script takes a GCS bucket name as an input argument, gets the lists of blobs in that bucket, prints the list of blobs and also writes them to an output file."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "# Create folders if they don't exist.\n",
+    "mkdir -p tmp/reuse_components_pipeline/mnist_training\n",
+    "\n",
+    "# Create the Python file that lists GCS blobs.\n",
+    "cat > ./tmp/reuse_components_pipeline/mnist_training/app.py <<HERE\n",
+    "import argparse\n",
+    "from datetime import datetime\n",
+    "import tensorflow as tf\n",
+    "\n",
+    "parser = argparse.ArgumentParser()\n",
+    "parser.add_argument(\n",
+    "    '--model_path', type=str, required=True, help='Name of the model file.')\n",
+    "parser.add_argument(\n",
+    "    '--bucket', type=str, required=True, help='GCS bucket name.')\n",
+    "args = parser.parse_args()\n",
+    "\n",
+    "bucket=args.bucket\n",
+    "model_path=args.model_path\n",
+    "\n",
+    "model = tf.keras.models.Sequential([\n",
+    "  tf.keras.layers.Flatten(input_shape=(28, 28)),\n",
+    "  tf.keras.layers.Dense(512, activation=tf.nn.relu),\n",
+    "  tf.keras.layers.Dropout(0.2),\n",
+    "  tf.keras.layers.Dense(10, activation=tf.nn.softmax)\n",
+    "])\n",
+    "\n",
+    "model.compile(optimizer='adam',\n",
+    "              loss='sparse_categorical_crossentropy',\n",
+    "              metrics=['accuracy'])\n",
+    "\n",
+    "print(model.summary())    \n",
+    "\n",
+    "mnist = tf.keras.datasets.mnist\n",
+    "(x_train, y_train),(x_test, y_test) = mnist.load_data()\n",
+    "x_train, x_test = x_train / 255.0, x_test / 255.0\n",
+    "\n",
+    "callbacks = [\n",
+    "  tf.keras.callbacks.TensorBoard(log_dir=bucket + '/logs/' + datetime.now().date().__str__()),\n",
+    "  # Interrupt training if val_loss stops improving for over 2 epochs\n",
+    "  tf.keras.callbacks.EarlyStopping(patience=2, monitor='val_loss'),\n",
+    "]\n",
+    "\n",
+    "model.fit(x_train, y_train, batch_size=32, epochs=5, callbacks=callbacks,\n",
+    "          validation_data=(x_test, y_test))\n",
+    "\n",
+    "from tensorflow import gfile\n",
+    "\n",
+    "gcs_path = bucket + \"/\" + model_path\n",
+    "# The export require the folder is new\n",
+    "if gfile.Exists(gcs_path):\n",
+    "    gfile.DeleteRecursively(gcs_path)\n",
+    "tf.keras.experimental.export_saved_model(model, gcs_path)\n",
+    "\n",
+    "with open('/output.txt', 'w') as f:\n",
+    "  f.write(gcs_path)\n",
+    "HERE"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create a Docker container\n",
```

Comment:
Please apply the same comments as previous notebooks.

Reply:
This has been changed to align with previous notebooks.